cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -24,68 +24,149 @@ import (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// MasterConfiguration contains all necessary information for kubeadm to bootstrap a cluster
 type MasterConfiguration struct {
-	metav1.TypeMeta
-
-	API                API
-	Etcd               Etcd
-	Networking         Networking
-	KubernetesVersion  string
-	CloudProvider      string
-	NodeName           string
-	AuthorizationModes []string
+	metav1.TypeMeta `json:",inline"`
+
+	// APIServer specifies how the api server component is set up and configured
+	APIServer APIServer `json:"apiServer"`
```

Comment:
these are purely internal types, aren't they? No need for any json tags here then.